src/signalling/room.rs

```diff
@@ -407,25 +445,32 @@ impl Handler<RpcConnectionEstablished> for Room {
     }
 }
 
-/// Signal of close [`Room`].
-#[derive(Debug, Message)]
-#[rtype(result = "()")]
-#[allow(clippy::module_name_repetitions)]
-pub struct CloseRoom {}
-
-impl Handler<CloseRoom> for Room {
-    type Result = ();
+// Close room on `SIGINT`, `SIGTERM`, `SIGQUIT`, `SIGHUP` signals.
+impl Handler<ShutdownMessage> for Room {
+    type Result = ActFuture<(), ()>;
 
-    /// Sends to remote [`Member`] the [`Event`] about [`Peer`] removed.
-    /// Closes all active [`RpcConnection`]s.
     fn handle(
         &mut self,
-        _msg: CloseRoom,
+        _: ShutdownMessage,
         ctx: &mut Self::Context,
     ) -> Self::Result {
-        info!("Closing Room [id = {:?}]", self.id);
-        let drop_fut = self.participants.drop_connections(ctx);
-        ctx.wait(wrap_future(drop_fut));
+        info!("Shut down signal received for Room: {:?}", self.id);
+
+        self.get_close_room_fut(ctx)
+    }
+}
+
+/// Signal to change [`Room`]'s state to closed.
+#[derive(Message)]
+#[rtype(result = "()")]
+#[allow(clippy::module_name_repetitions)]
+struct RoomSetStopped {}
```

Comment:
@tyranron,

It was used to set `Room`'s state to `Stopped` back when I was returning futures from `ShutdownGracefully` because I couldn't modify self from these futures